src/classes/AddressService.cls

```diff
@@ -114,16 +116,70 @@ public inherited sharing class AddressService {
     }
 
     /*******************************************************************************************************
-    * @description utility to combine mailing street fields into a single string with embedded CR LF
-    * @param addr the Address to look at
-    * @return boolean
+    * @description Utility to copy Address fields from an Address object to a Contact or Account.
+    * Handles instances where State and Country picklists are enabled, and multiline street addresses.
+    * @param addr the Address object to copy from
+    * @param sobjDst the destination Contact or Account
+    * @param strFieldPrefix the address fields to copy to, ie., Mailing, Other, Shipping, Billing
+    * @param strFieldAddrType an optional Address Type field on sobjDst to copy to
+    * @return void
     */
-    public String getMultilineStreet(Address__c addr) {
-        Boolean isOneLineStreetAddress = String.isBlank(addr.MailingStreet2__c);
-        if (isOneLineStreetAddress) {
-            return addr.MailingStreet__c;
+    public void copyOntoSObject(IAddress anAddress, SObject sobjDst,
+            String strFieldPrefix, String strFieldAddrType) {
+        Address__c addr = (Address__c) anAddress.getRecord();
```

Comment:
Yes, @ptewson-sfdo spot on!  This should definitely be a follow up item also.  This method body should really be an implementation detail on each of the two entities Contact and Household. When you give them each an Address, they know how to set their fields internally, instead of here in this AddressService class.  But I am going to reserve this one for a follow up unless I hear otherwise!

Reply:
And yes @voduyemi, as it is this method would be better named something like buildContactAndAccountAddresses and have a guard that only accepts the two supported SObjects.  But, ultimately it should be broken up and relocated inside the entities themselves, and then it won't be a method that supports multiple SObjects.